src/figuresClasses.ts

```diff
@@ -1,19 +1,87 @@
+type FigureShape = 'triangle' | 'circle' | 'rectangle';
+type FigureColor = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: FigureShape;
+  color: FigureColor;
+  getArea(): number;
 
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: FigureShape;
 
-}
+  constructor(
+    public color: FigureColor,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Length of sides must be greater than 0.');
+    }
+
+    if (a + b <= c || b + c <= a || c + a <= b) {
+      throw new Error('The longest side must be greater than others side.');
+    }
+
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
```

Comment:
`public` word already did it, it's ts feature

Reply:
'constructor(public color: FigureColor)'

it's the same as 

'
color: FigureColor;

constructor(color: FigureColor) {
 this.color = color;
}
'

Reply:
please check this article https://kendaleiv.com/typescript-constructor-assignment-public-and-private-keywords/